src/circuit.rs

```diff
@@ -139,17 +197,28 @@ impl<'r, F: Field> Region<'r, F> {
         column: Column<Advice>,
         offset: usize,
         mut to: V,
-    ) -> Result<Cell, Error>
+    ) -> Result<AssignedCell<VR, F>, Error>
     where
         V: FnMut() -> Result<VR, Error> + 'v,
-        VR: Into<Assigned<F>>,
+        for<'vr> Assigned<F>: From<&'vr VR>,
         A: Fn() -> AR,
         AR: Into<String>,
     {
-        self.region
-            .assign_advice(&|| annotation().into(), column, offset, &mut || {
-                to().map(|v| v.into())
-            })
+        let mut value = None;
+        let cell =
+            self.region
+                .assign_advice(&|| annotation().into(), column, offset, &mut || {
```

Comment:
Why isn't the `annotation` just passed in directly in these sorts of places, instead of wrapping it in another closure?

Reply:
Lifetimes IIRC.